packages/std/src/coins.rs

```diff
@@ -270,6 +270,12 @@ impl DoubleEndedIterator for CoinsIntoIter {
     }
 }
 
+impl ExactSizeIterator for CoinsIntoIter {
```

Comment:
I think for this we also need to change the size_hint implementation:

> When implementing an ExactSizeIterator, you must also implement [Iterator](https://doc.rust-lang.org/std/iter/trait.Iterator.html). When doing so, the implementation of [Iterator::size_hint](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.size_hint) must return the exact size of the iterator.

Reply:
Good catch!